### ISSUE TITLE
Godot can get stuck on load (some sort of a filesystem/cache problem)

### ISSUE THREAD

**Issue by Zireael07** [godotengine/godot#63615]
### Godot version

4.0 a10&a13

### System information

Linux Manjaro, Vulkan, Intel Kaby Lake

### Issue description

Hit a weird issue today. Did some changes yesterday that didn't work. Reverted them in git (while Godot itself was open, that was probably silly of me but I did that with no problems in 3.x) and I ended up with a project that could be run in a10 (so the scripts/assets weren't broken) but NOT opened in editor.

Editor window was non-functional (filesystem tab empty, top bar - Scene/Project - and bottom bar - output/animation - completely not working, X to close not working). Had to xkill Godot.

@smix8 helped me figure out that I had to nuke .godot (everything, after nuking just the imported it froze after reimport). He also says that this is NOT a one-off issue. Can we please: 
a) have at least the X functional?
b) have verbose tell us something happened and how to fix it ("Please clear editor cache in .godot") and/or something in the editor to the same effect?

### Steps to reproduce

???

### Minimal reproduction project

N/A

**Comment by fisher2470** [comment]
Is this related to the filesystem update issue here: https://github.com/godotengine/godot/issues/57870?

Seems like the HashMap implementation might be causing issues with the filesystem whenever changes are made during execution?

**Comment by Zireael07** [comment]
No clue as I never had that find->replace problem and repeated reopening didn't help (I tried multiple times)

**Comment by Calinou** [comment]
Can you reproduce this after enabling single-window mode in the Editor Settings?

**Comment by Zireael07** [comment]
Forgot to mention that I always run with --single-window parameter because I just don't want to deal with weird stuff that multi window mode has on KDE (e.g. focus getting weird)

**Comment by Zireael07** [comment]
Just occurred again, after reverting/discarding some changes in Git. This time, with Godot CLOSED.

Just to check if I was right, this time I nuked only the godot_uid.bin and the editor subfolder of .godot (leaving imported and shader_cache untouched). So this means this is either the editor stuff or the godot_uid causing this.
This allowed the project to launch again.

**Comment by Zireael07** [comment]
Happened again, in a14, after reverting/discarding changes in Git with Godot closed.

It's NOT godot_uid. I tried to find out WHAT it is by deleting the most likely culprits (filesystem_cache, the cfg file relating to the file that had its changes reverted) but failed. Nuking the contents of .godot/editor made the editor launch again.

**Comment by Zireael07** [comment]
![Screenshot_20200121_125016](https://user-images.githubusercontent.com/4525388/188271506-0c580023-fdc2-442c-9f63-f2a9eb44969e.png)

That's as far as I was able to go with VS Code and debugger.

**Comment by Zireael07** [comment]
Ok, I was able to dig a little deeper - it gets stuck while reopening scripts, so as a temporary fix disabling restore scripts on load in settings should also work. Still trying to figure out where it dies and why the resource seems to end up in limbo, never opened/returned and therefore Godot is stuck

**Comment by Zireael07** [comment]
Updated OP with instructions on how to get a reproducer.

**Comment by Zireael07** [comment]
Yep, disabling restore scripts on load fixed the problem (or rather worked around it). See the proposal asking about being able to change editor settings from project manager, though - I had to open another project to do it

**Comment by Calinou** [comment]
@Zireael07 Can you reproduce this on 4.0.beta7?
